### PR TITLE
Use esbuild to minify frontend JS

### PR DESCRIPTION
**Pull request by davidmehren**
### Component/Part
Webpack config

### Description
This PR switches the minifier in Webpack from https://github.com/terser/terser to https://esbuild.github.io. esbuild is written in Go and reduces the build time from `54s wall/198s user` to `12s wall/21s user` on my machine. 

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -2,8 +2,7 @@
   "presets": [
     ["env", {
       "targets": {
-        "node": "8",
-        "uglify": true
+        "node": "8"
```
Comment (ErikMichelson) [inline .babelrc:5]: Why do we target node 8, if we raised the minimum node version to 12?

Comment (davidmehren) [reply .babelrc:5]: I have no idea. It doesn't even make sense to target Node, as Babel only runs on code for the frontend.

**Review by mrdrogdrog — APPROVED** [review]

File: .babelrc (hunk 1 above)
Comment (davidmehren) [reply .babelrc]: I changed the Babel settings to `browsers: defaults`, which equals `> 0.5%, last 2 versions, Firefox ESR, not dead`, which seems more reasonable.
Not sure if that had any effect on bundle size...

**Review by InnayTool — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -179,6 +179,7 @@
     "babel-runtime": "6.26.0",
     "copy-webpack-plugin": "6.4.1",
     "css-loader": "5.2.4",
+    "esbuild-loader": "^2.12.0",
```
Comment (DerMolly) [inline package.json:182]: This should be pinned…

**Review by DerMolly — CHANGES_REQUESTED** [review]

**Review by mrdrogdrog — DISMISSED** [review]

**Review by ErikMichelson — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (davidmehren) [reply package.json]: Pinned!

**Review by mrdrogdrog — APPROVED** [review]

**Review by DerMolly — APPROVED** [review]

**Review by InnayTool — APPROVED** [review]

**Pull request merged**